src/services/EdFi.DataManagementService.Api/Core/Middleware/ParsePathMiddleware.cs

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+using System.Text.RegularExpressions;
+using EdFi.DataManagementService.Api.Core.Model;
+using EdFi.DataManagementService.Core.Pipeline;
+
+namespace EdFi.DataManagementService.Api.Core.Middleware;
+
+/// <summary>
+/// Parses and validates the path from the frontend is well formed. Adds PathComponents
+/// to the context if it is.
+/// </summary>
+public partial class ParsePathMiddleware(ILogger _logger) : IPipelineStep
+{
+    // Matches all of the following sample expressions:
+    // /ed-fi/Sections
+    // /ed-fi/Sections/
+    // /ed-fi/Sections/idValue
+    [GeneratedRegex(@"\/(?<projectNamespace>[^/]+)\/(?<endpointName>[^/]+)(\/|$)((?<documentUuid>[^/]*$))?")]
```

Comment:
https://learn.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.generatedregexattribute?view=net-8.0

If I'm reading that correctly, this successfully gives us a pre-compiled regex ✅ .

Reply:
Yes, this seemed the "modern" way to do it.